python/cudf/cudf/core/series.py

```diff
@@ -2626,6 +2630,38 @@ def _align_to_index(
         result = lhs.join(rhs, how=how, sort=sort)[0]
         return result
 
+    def merge(self, other):
+
+        dummy = "name"
+        l_name = self.name
+        r_name = other.name
+
+        if l_name is None and r_name is not None:
+            self.name = r_name
+            left_on = right_on = r_name
+        if r_name is None and l_name is not None:
+            other.name = l_name
```

Comment:
You're modifying other inplace here. If you want to do this you need to make a shallow copy first.

Reply:
Updated